luarules/gadgets/game_tax_resource_sharing.lua

```diff
@@ -28,55 +28,93 @@ local spGetTeamUnitCount = Spring.GetTeamUnitCount
 local gameMaxUnits = math.min(Spring.GetModOptions().maxunits, math.floor(32000 / #Spring.GetTeamList()))
 
 local sharingTax = Spring.GetModOptions().tax_resource_sharing_amount
+local metalTaxThreshold = Spring.GetModOptions().player_metal_send_threshold or 0 -- Use standardized key
```

Comment:
I am not against adding more changes to the same PR but I don't think it's a good idea here, makes harder to review and manage rollbacks.

Reply:
Ok I will open a separate PR.

Reply:
Alright, I'll talk from my end. Unless I find mana to revisit and review with these changes I won't do so for now. Unless strictly related functionality I recommend separate PRs. Feel free to resolve this comment.

Reply:
Ah screw it I'll just split it lol

Reply:
done https://github.com/beyond-all-reason/Beyond-All-Reason/pull/4804